src/pages/docs/ai-transport/features/token-streaming/message-per-response.mdx

```diff
@@ -0,0 +1,373 @@
+---
+
+Stream LLM and generative AI responses efficiently by appending individual tokens to a single
+message on an Ably channel. This pattern creates one complete response message in channel history
+while delivering tokens in realtime.
+
+## Overview
+
+The message-per-response pattern enables you to stream AI-generated content as individual tokens in
+realtime, while maintaining a clean, compacted message history. Each AI response becomes a single
+message that grows as tokens are appended, resulting in efficient storage and easy retrieval of
+complete responses.
+
+### When to use this pattern
+
+This approach is ideal when:
+
+- You want each complete AI response stored as a single message in history.
+- You want clients joining mid-stream to catch up efficiently without processing thousands of
+  individual tokens.
+- Your application displays progressive AI responses that build up over time.
+
+### How it works
+
+1. **Initial message**: When an AI response begins, publish an initial message with `message.create`
+   action to the Ably channel with an empty or the first token as content.
+2. **Token streaming**: Append subsequent tokens to the original message by publishing those tokens
+   with the `message.append` action.
+3. **Live Delivery**: Clients subscribed to the channel receive each appended token in realtime, allowing
+   them to progressively render the response.
+4. **Compacted history**: The channel history contains only one message per AI response,
+   which includes all tokens appended to it concatenated together.
+
+You do not need to mark the message or token stream as completed; the final message will
+automatically have the full response with all tokens appended to it.
+
+## Publishing tokens
+
+Publish tokens from a [Realtime](/docs/api/realtime-sdk) client, which maintains a
+persistent connection to the Ably service. This allows you to publish at very high message rates
+with the lowest possible latencies, while preserving guarantees around message delivery order.
+For more information, see [Realtime and REST](/docs/basics#realtime-and-rest).
+
+[Channels](/docs/channels) separate message traffic into different topics.
+For token streaming, each conversation or session typically has its own channel.
+
+Use the [`get()`](/docs/api/realtime-sdk/channels#get) method to create or retrieve a channel instance:
+
+<Code>
+```javascript
+const channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}');
+```
+</Code>
+
+
+To start streaming an AI response, publish the initial message. Then append each subsequent token
+to that message as it arrives from the AI model:
+<Code>
+```javascript
+// set the initial data to empty string or the first token
+// capture the msgSerial for appending subsequent tokens
+
+const { serials: [msgSerial] } = await channel.publish('ai-response', { data: '' });
+
+for await (const token of streamedResponse) {
```

Comment:
I think we should include a comment indicating what this example assumes is returned from the stream, like we do in https://github.com/ably/docs/pull/3014/changes#diff-b66bbc76507fb59b061f92037a82a9884804c3668f11a277fb5495b6a3819deb

Also *nit*, but we should use a consistent variable name for the stream, it is called `stream` elsewhere